src/api/java/baritone/api/Settings.java

```diff
@@ -1385,14 +1421,82 @@ public final Type getType() {
         settingTypes = Collections.unmodifiableMap(tmpSettingTypes);
     }
 
+    /**
+     * Register a new setting.
+     *
+     * @param name         the name of the setting
+     * @param defaultValue the default value of the setting
+     * @param <T>          the type of the setting value
+     * @return the newly registered setting
+     */
+    public <T> Setting<T> registerSetting(String name, T defaultValue) {
+        Setting<T> setting = new Setting<>(defaultValue);
+        setting.name = name;
+        return superSecretRegisterSetting(setting, name.toLowerCase(), defaultValue.getClass());
+    }
+
+    /**
+     * Register settings from the fields of the given class.
+     *
+     * @param inputClass the class containing the settings fields
+     */
+    public void registerSettingsFromClass(Class<?> inputClass) {
+        Field[] fields = inputClass.getDeclaredFields();
+
+        for (Field field : fields) {
+            if (field.getType().equals(Setting.class)) {
+                try {
+                    field.setAccessible(true);
+                    Setting<?> setting = (Setting<?>) field.get(this);
+                    String name = field.getName();
+                    setting.name = name;
+                    String lowercaseName = name.toLowerCase();
+                    superSecretRegisterSetting(setting, lowercaseName, ((ParameterizedType) field.getGenericType()).getActualTypeArguments()[0]);
+                } catch (IllegalAccessException e) {
+                    throw new IllegalStateException("Error accessing setting field", e);
+                }
+            }
+        }
+    }
```

Comment:
This relies on the fields of the input class having the required type information, which is not guaranteed and can break other people's code in unexpected ways.
If there is a comment "here be dragons" it is probably not a good idea to expose exactly that code as public API.

This fails if the field is null.